lib/server/src/digest.rs

```diff
@@ -0,0 +1,52 @@
+use failure::{self, Error};
+use std::io::Read;
+
+// Crypto and crypto related imports
+use sha2::{Sha256, Digest};
+
+// Buffer size for SHA2 hashing
+const BUFFER_SIZE: usize = 1024;
+
+fn digest<D: Digest + Default, R: Read>(reader: &mut R) -> Result<String, Error> {
```

Comment:
thanks for pulling this out to a separate file, it helps.

Reply:
No problem.  
Since I was moving away from rust-crypto crate, I thought it was a good idea to adjust a bit the digest part and write tests to make sure the digest was calculated correctly.